src/js/wrap.cpp

```diff
@@ -489,6 +489,52 @@ namespace js
     return JS_UNDEFINED;
   }
 
+  JSValue js_network_generate_endorsed_certificate(
+    JSContext* ctx,
+    JSValueConst this_val,
+    int argc,
+    [[maybe_unused]] JSValueConst* argv)
+  {
+    if (argc != 1)
+    {
+      return JS_ThrowTypeError(ctx, "Passed %d arguments but expected 1", argc);
+    }
+
+    auto network =
+      static_cast<ccf::NetworkState*>(JS_GetOpaque(this_val, network_class_id));
+    if (network == nullptr)
+    {
+      return JS_ThrowInternalError(ctx, "Network state is not set");
+    }
+
+    auto global_obj = Context::JSWrappedValue(ctx, JS_GetGlobalObject(ctx));
+    auto ccf =
+      Context::JSWrappedValue(ctx, JS_GetPropertyStr(ctx, global_obj, "ccf"));
+    auto node_ =
+      Context::JSWrappedValue(ctx, JS_GetPropertyStr(ctx, ccf, "node"));
+
+    auto node =
+      static_cast<ccf::AbstractNodeState*>(JS_GetOpaque(node_, node_class_id));
+
+    if (node == nullptr)
+    {
+      return JS_ThrowInternalError(ctx, "Node state is not set");
+    }
+
+    auto csr_cstr = JS_ToCString(ctx, argv[0]);
```

Comment:
This needs to be wrapped in an auto_free, otherwise we'll leak the cstring when we throw on 527.

Reply:
Yep. The pattern seems to be consistent across this file and I'm planning to address it in the separate PR (see #2910).